File: platform/polyglot/extract-imports-text-based-notation/go/specifier.go
package extractimports

import "fmt"

// ExtractedImport represents a module and its specifiers in text-based notation.
type ExtractedImport struct {
	ModuleName string
	Specifiers []string
}

// SpecifierKind constants for specifier type prefixes.
const (
	KindDefault     = "default"
	KindNamed       = "named"
	KindNamespace   = "namespace"
	KindSideEffect  = "side-effect"
	KindExportNamed = "export-named"
	KindExportAll   = "export-all"
)

func FormatNamed(original, alias string) string {
	if alias != "" && alias != original {
		return fmt.Sprintf("named: %s as %s", original, alias)
	}
	return fmt.Sprintf("named: %s", original)
}

func FormatNamespace(name string) string {
	return fmt.Sprintf("namespace: %s", name)
}

func FormatSideEffect(name string) string {
	return fmt.Sprintf("side-effect: %s", name)
}

func FormatExportNamed(name string) string {
	return fmt.Sprintf("export-named: %s", name)
}

File: platform/polyglot/extract-imports-text-based-notation/rs/src/specifier.rs
use std::fmt;

#[derive(Debug, Clone, PartialEq)]
pub enum SpecifierKind {
    Default,
    Named,
    Namespace,
    SideEffect,
    ExportDefault,
    ExportNamed,
    ExportAll,
    ExportNamespace,
}

impl fmt::Display for SpecifierKind {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            Self::Default => write!(f, "default"),
            Self::Named => write!(f, "named"),
            Self::Namespace => write!(f, "namespace"),
            Self::SideEffect => write!(f, "side-effect"),
            Self::ExportDefault => write!(f, "export-default"),
            Self::ExportNamed => write!(f, "export-named"),
            Self::ExportAll => write!(f, "export-all"),
            Self::ExportNamespace => write!(f, "export-namespace"),
        }
    }
}

pub fn format_named(name: &str, alias: Option<&str>) -> String {
    match alias {
        Some(a) if a != name => format!("named: {} as {}", name, a),
        _ => format!("named: {}", name),
    }
}

pub fn format_namespace(name: &str) -> String {
    format!("namespace: {}", name)
}

pub fn format_export_named(name: &str, alias: Option<&str>) -> String {
    match alias {
        Some(a) if a != name => format!("export-named: {} as {}", name, a),
        _ => format!("export-named: {}", name),
    }
}

pub fn format_export_all() -> String {
    "export-all".to_string()
}
